lib/IRGen/IRGenSIL.cpp

```diff
@@ -2260,9 +2260,11 @@ void IRGenSILFunction::visitFloatLiteralInst(swift::FloatLiteralInst *i) {
 static llvm::Constant *getAddrOfString(IRGenModule &IGM, StringRef string,
                                        StringLiteralInst::Encoding encoding) {
   switch (encoding) {
-  case swift::StringLiteralInst::Encoding::UTF8:
-    return IGM.getAddrOfGlobalString(string);
-
+  case swift::StringLiteralInst::Encoding::UTF8: {
+    return IGM.getAddrOfGlobalString(string,
+                                     /*willBeRelativelyAddressed=*/false,
+                                     /*addNull=*/false);
```

Comment:
I think that's also better handled by adding a `getAddrOfNullTerminatedGlobalString` helper that adds the null terminator then calls `getAddrOfGlobalString`. As written, you're caching only keyed by the input `StringRef`, so this would break:

```
getAddrOfGlobalString("foo", false, false);
getAddrOfGlobalString("foo", false, true); // will return unterminated 'foo' from cache
```